scripts/core/EXP-COMMAND-BUFER.rst

```diff
@@ -0,0 +1,116 @@
+<%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+.. _experimental-command-buffer:
+
+==============
+Command-Buffer
+==============
+
+.. warning::
+
+    Experimental features:
+
+    *   May be replaced, updated, or removed at any time.
+    *   Do not require maintaining API/ABI stability of their own additions
+        over time.
+    *   Do not require conformance testing of their own additions.
+
+
+A command-buffer represents a series of commands for execution on a command queue.
+Many adapters support this kind of construct either natively or through extensions,
+but they are not available to use directly. Typically their use is abstracted through
+the existing Core APIs, for example when calling ${x}EnqueueKernelLaunch the adapter
+may both append the kernel command to a command-buffer-like construct and also submit
+that command-buffer to a queue for execution. These types of structures allow for
+batching of commands to improve host launch latency, but without direct control it
+falls to the adapter implementation to implement automatic batching of commands.
+
+This experimental feature exposes command-buffers in the Unified Runtime API directly,
+allowing applications explicit control over the enqueue and execution of commands to
+batch commands as required for optimal performance.
+
+Querying Command-Buffer Support
+===============================
+
+Support for command-buffers can be queried for a given device/adapter by using the
+device info query with ${X}_DEVICE_INFO_EXTENSIONS. Adapters supporting this experimental
+feature will report ${X}_COMMAND_BUFFER_EXTENSION_STRING_EXP in the returned list of
```

Comment:
While the macro is a nice helper for the user the way this is worded suggests that the string `UR_COMMABD_BUFFER_EXTENSION_STRING_EXP` will appear in the returned string. I feel like this sentence should instead be using the actual string the user should expect to see, I feel like the macro should be explicitly called a macro and mentioned in a *hint* block.

Reply:
I agree the text could probably better reflect this is a macro and not the string itself. I did have the actually string here originally but I figured replacing it with a defined value would avoid any potential mismatch in the future if the string changes. Perhaps that's not worth worrying about in this singular case though.

I think that #592 would help with this problem, but perhaps including the actual string value here is the best option for now.

Reply:
I'm not against the macro existing for ease of use but I do think this should actually document the expected string being returned from `urDeviceGetInfo`.

Reply:
That's fair! I've made the suggested changes now.